README.md

```diff
@@ -1,6 +1,826 @@
+
+<details markdown='1'><summary>Solution Code</summary>
+
+```solidity
+    function calculateCollateralValue(address user) public view returns (uint256) {
+        uint256 collateralAmount = s_userCollateral[user]; // Get user's collateral amount
+        return (collateralAmount * i_cornDEX.currentPrice()) / 1e18; // Calculate collateral value in CORN
+    }
+```
+
+</details>
+</details>
+
+🔄 Let's turn our attention to the internal `_calculatePositionRatio` view function.
+
+📊 This function takes a user address and returns what we are calling the "position ratio". This is the percentage of collateral to borrowed assets with a caveat, it is returned as the percentage * 1e18. In other words, if the collateral ratio percent is 133 then the returned value would be 133000000000000000000. We do this to enable a higher amount of precision. Try to figure out the math on your own.
+
+<details markdown='1'><summary>🔎 Hint</summary>
+
+> 📝 It needs to return the value of the collateral (in CORN) divided by the amount borrowed.
+
+<details markdown='1'><summary>Solution Code</summary>
+
+```solidity
+    function _calculatePositionRatio(address user) internal view returns (uint256) {
+        uint borrowedAmount = s_userBorrowed[user]; // Get user's borrowed amount
+        uint collateralValue = calculateCollateralValue(user); // Calculate user's collateral value
+        if (borrowedAmount == 0) return type(uint256).max; // Return max if no corn is borrowed
+        return (collateralValue * 1e18) / borrowedAmount; // Calculate position ratio
+    }
+```
+
+</details>
+</details>
+
+🫗 Now we will fill in the details on `isLiquidatable`. This function should return a bool indicating if the position ratio is less than `COLLATERAL_RATIO`. See if you can implement the logic without the hint.
+
+<details markdown='1'><summary>🔎 Hint</summary>
+
+> 🎯 We can use the `_calculatePositionRatio` function we just made to get the current ratio. Then just a simple comparison between that and the COLLATERAL_RATIO to make sure we aren't below the acceptable liquidatable threshold.
+
+<details markdown='1'><summary>Solution Code</summary>
+
+```solidity
+    function isLiquidatable(address user) public view returns (bool) {
+        uint256 positionRatio = _calculatePositionRatio(user); // Calculate user's position ratio
+        return (positionRatio * 100) < COLLATERAL_RATIO * 1e18; // Check if position is unsafe
+    }
+```
+
+</details>
+</details>
+
+✅ Lastly let's fill in a simple function called `_validatePosition`. This function has one use case: revert with `Lending_UnsafePositionRatio` if the user's position is liquidatable (`isLiquidatable` returns exactly what we need). We can then use this function any place where we need to verify the user's ratio position hasn't been changed to a liquidatable state after updating the user's state.
```

Comment:
I think it's better to add hint/solution for `_validatePosition` for consistency

Here and for other Readme changes add it to second readme too, please